src/Stats.PostProcessReports/DetailedReportPostProcessor.cs

```diff
@@ -0,0 +1,371 @@
+
+            if (!sourceBlobs.Where(b => GetBlobName(b) == SuccessFilename).Any())
+            {
+                _logger.LogInformation("No " + SuccessFilename + " file present in source location, terminating until file is available.");
+                return;
+            }
+
+            var copySucceeded = await _workStorage.ExistsAsync(CopySucceededFilename, cancellationToken);
+            var jobSucceeded = await _workStorage.ExistsAsync(JobSucceededFilename, cancellationToken);
+
+            var sourceJsonBlobs = sourceBlobs.Where(b => b.Uri.AbsolutePath.EndsWith(JsonExtension, StringComparison.OrdinalIgnoreCase)).ToList();
+            var copyNeeded = true;
+            var sourceBlobsExist = false;
+            if (copySucceeded)
+            {
+                sourceBlobsExist = await SourceBlobExistsInWorkLocationAsync(sourceJsonBlobs, cancellationToken);
+                copyNeeded = !sourceBlobsExist;
+            }
+
+            if (copyNeeded)
+            {
+                await CopySourceBlobsAsync(sourceJsonBlobs, copySucceeded, cancellationToken);
+            }
+            else
+            {
+                _logger.LogInformation("Previous copy succeeded, will not copy.");
+            }
+
+            if (copySucceeded && jobSucceeded && sourceBlobsExist)
+            {
+                _logger.LogInformation("No new data, terminating until updated.");
+                return;
+            }
+
+            var workBlobs = await EnumerateWorkStorageBlobsAsync();
+            var workJsonBlobs = workBlobs.Where(b => b.Uri.AbsolutePath.EndsWith(JsonExtension, StringComparison.OrdinalIgnoreCase)).ToList();
+            await ProcessBlobs(workJsonBlobs, cancellationToken);
+
+            _logger.LogInformation("Done processing");
+        }
+
+        private async Task ProcessBlobs(List<StorageListItem> jsonBlobs, CancellationToken cancellationToken)
+        {
+            var totals = new TotalStats();
+            foreach (var sourceBlob in jsonBlobs)
+            {
+                var blobName = GetBlobName(sourceBlob);
+                var workBlobUri = _workStorage.ResolveUri(blobName);
+                var sourceBlobStats = new BlobStatistics();
+                var individualReports = await ProcessSourceBlobAsync(sourceBlob, sourceBlobStats, totals);
+                using (_logger.BeginScope("Processing {BlobName}", blobName))
+                {
+                    if (individualReports != null && individualReports.Any())
+                    {
+                        var consumerTasks = Enumerable
+                            .Range(1, _configuration.ReportWriteDegreeOfParallelism)
+                            .Select(instanceId => WriteReports(instanceId, individualReports, sourceBlobStats, blobName, cancellationToken))
+                            .ToList();
+
+                        await Task.WhenAll(consumerTasks);
+                        ++totals.SourceFilesProcessed;
```

Comment:
```suggestion
                        totals.SourceFilesProcessed++;
```
🤬 (just kidding, feel free to ignore)